pyuvdata/telescopes.py

```diff
@@ -71,21 +90,163 @@
 }
 
 
+def known_telescopes():
+    """
+    Get list of known telescopes.
+
+    Returns
+    -------
+    list of str
+        List of known telescope names.
+    """
+    astropy_sites = [site for site in EarthLocation.get_site_names() if site != ""]
+    known_telescopes = list(set(astropy_sites + list(KNOWN_TELESCOPES.keys())))
+    return known_telescopes
+
+
+def known_telescope_location(
+    name: str,
+    return_citation: bool = False,
+    known_telescope_dict: dict = KNOWN_TELESCOPES,
+):
+    """
+    Get the location for a known telescope.
+
+    Parameters
+    ----------
+    name : str
+        Name of the telescope
+    return_citation : bool
+        Option to return the citation.
+    known_telescope_dict: dict
+        This should only be used for testing. This allows passing in a
+        different dict to use in place of the KNOWN_TELESCOPES dict.
+
+    Returns
+    -------
+    location : EarthLocation
+        Telescope location as an EarthLocation object.
+    citation : str, optional
+        Citation string.
+
+    """
+    astropy_sites = EarthLocation.get_site_names()
+    telescope_keys = list(known_telescope_dict.keys())
+    telescope_list = [tel.lower() for tel in telescope_keys]
+
+    # first deal with location.
+    if name in astropy_sites:
+        location = EarthLocation.of_site(name)
+
+        citation = "astropy sites"
+    elif name.lower() in telescope_list:
+        telescope_index = telescope_list.index(name.lower())
+        telescope_dict = known_telescope_dict[telescope_keys[telescope_index]]
+        citation = telescope_dict["citation"]
+
+        if telescope_dict["center_xyz"] is not None:
+            location = EarthLocation.from_geocentric(
+                *telescope_dict["center_xyz"], unit="m"
+            )
+        else:
+            if (
+                telescope_dict["latitude"] is None
+                or telescope_dict["longitude"] is None
+                or telescope_dict["altitude"] is None
+            ):
+                raise ValueError(
+                    "Bad location information in known_telescopes_dict "
+                    f"for telescope {name}. Either the center_xyz "
+                    "or the latitude, longitude and altitude of the "
+                    "telescope must be specified."
+                )
+            location = EarthLocation.from_geodetic(
+                lat=telescope_dict["latitude"] * units.rad,
+                lon=telescope_dict["longitude"] * units.rad,
+                height=telescope_dict["altitude"] * units.m,
+            )
```

Comment:
why isn't `KNOWN_TELESCOPES` just a dictionary of `{name: Telescope()}` objects?

Reply:
ah, is it because `Telescope` has more information in it that's required than can be reasonably specified here? In that case, at least I would use a `location` parameter in the dict, which is simply an astropy Location.